### ISSUE TITLE
Getting 'undefined' error when adding DigitalOcean node

### ISSUE THREAD

**Issue by jooosh** [rancher/rancher#11280]
In the process of creating a new RKE cluster, when adding new nodes via DigitalOcean, an 'undefined' error is displayed.  I've confirmed that my access token is valid and was able to create droplets outside of Rancher using the access key.  I thought I saw the same issue during the January online meetup.  Any assistance would be appreciated.

![image](https://user-images.githubusercontent.com/356481/35750984-151790e2-0814-11e8-9a5b-7df44d7434a1.png)

---
| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v2.0.0-alpha12` UI: `v2.0.16` |
|Access|`local` `admin`|
|Route|`global-admin.clusters.new.rke`|

**Comment by tfiduccia** [comment]
@jooosh - I can't seem to reproduce your problem.  Can you please go into logs and see if there are errors there.

**Comment by jooosh** [comment]
@tfiduccia - Thanks for the help!  For the logs, you're referring to the Rancher Server container, correct?  Running docker logs on the container does not show any funky errors.  Is there some other log I should be looking at?

**Comment by tfiduccia** [comment]
@jooosh - 
1.  Can you create an empty RKE cluster, then go to Nodes and add a DO node that way? 
2.  While you are reproducing the error, can you have the dev tools open and under network see what the error is sending back.

**Comment by jooosh** [comment]
TLDR: Putting a Rancher server behind Cloudflare can cause issues.

@tfiduccia After having a weekend to disconnect, I had thought that maybe it was the fact that we had the docker server behind our CDN (Cloudflare).  We currently run Rancher 1.5.5 behind Cloudflare (for the GUI not for host communication) and we haven't had any issues.  However, I also tried to connect to the DigitalOcean API with this 1.5.5 instance and received the same error.

Bypassing Cloudflare resolved the issue.

Thanks again for the support.

**Comment by tfiduccia** [comment]
@jooosh - Happy you were able to figure it out.

**Comment by sixcorners** [comment]
It's not working because of this:
https://support.cloudflare.com/hc/en-us/articles/200171976-Error-1000-DNS-points-to-prohibited-IP
If the proxy could strip out the cf-ray header I think it would start working.

**Comment by sixcorners** [comment]
@tfiduccia Could this be reopened?

**Comment by tfiduccia** [comment]
Version - master 3/11
Verified fixed

**Comment by Kritika2206** [comment]
> TLDR: Putting a Rancher server behind Cloudflare can cause issues.
> 
> @tfiduccia After having a weekend to disconnect, I had thought that maybe it was the fact that we had the Rancher server behind our CDN (Cloudflare). We currently run Rancher 1.5.5 behind Cloudflare (for the GUI not for host communication) and we haven't had any issues. However, I also tried to connect to the DigitalOcean API with this 1.5.5 instance and received the same error.
> 
> Bypassing Cloudflare resolved the issue.
> 
> Thanks again for the support.

Hello @jooosh can you help with any document to setup rancher(Which is running on EKS) behind the cloudflare VPN.

Thanks you .
